### PR TITLE
Feature/ung 1358 groups

### PR DESCRIPTION
**Pull request by Carlos-Augusto**
- Adding support for groups
 - scopes 
 - sdk

### CODE REVIEW

File: sdk/src/main/scala/com/ubirch/defaults/DefaultExternalGetter.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.apache.http.entity.ByteArrayEntity
+
+@Singleton
+class DefaultExternalGetter @Inject() (config: Config, HMAC: HMAC) extends ExternalStateGetter {
+
+  private val httpclient: CloseableHttpClient = HttpClients.createDefault
+
+  private final val TOKEN_MANAGER_ENDPOINT: String = config.getString(Paths.TOKEN_SERVICE_PATH)
+
+  def execute(request: HttpUriRequest): ExternalResponseData[Array[Byte]] = {
```
Comment (Hiroki6) [inline sdk/src/main/scala/com/ubirch/defaults/DefaultExternalGetter.scala:22]: what happens when an error occurs in this method?
the error is thrown?

Comment (Hiroki6) [reply sdk/src/main/scala/com/ubirch/defaults/DefaultExternalGetter.scala:22]: Oh, I found where this method is called. it's wrapped in Task.
I prefer this method to return Task. it's safer.

File: src/main/scala/com/ubirch/models/Scopes.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  }
+}
+case object Anchor extends Action
+case object Verify extends Action
+case object Create extends Action
+case object GetInfo extends Action
+
+object Scopes {
+
+  type Scope = (Resource, Action)
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/models/Scopes.scala]: why don't use a case class like this?
it's readable for me.
```scala
case class Scope(resource: Resource, action: Action)
```

File: src/main/scala/com/ubirch/services/state/ExternalStateGetter.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+@Singleton
+class DefaultExternalGetter @Inject() (config: Config) extends ExternalStateGetter {
+
+  private val httpclient: CloseableHttpClient = HttpClients.createDefault
+
+  private final val DEVICE_GROUPS_ENDPOINT: String = config.getString(ExternalStateGetterPaths.DEVICE_GROUPS_ENDPOINT)
+  private final val TENANT_GROUPS_ENDPOINT: String = config.getString(ExternalStateGetterPaths.TENANT_GROUPS_ENDPOINT)
+
+  def execute(request: HttpUriRequest): ExternalResponseData[Array[Byte]] = {
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/state/ExternalStateGetter.scala:26]: here is same as well

File: src/main/scala/com/ubirch/models/Scopes.scala (hunk 2 above)
Comment (Carlos-Augusto) [reply src/main/scala/com/ubirch/models/Scopes.scala]: Yes, this is more readable. Added suggestion.

File: sdk/src/main/scala/com/ubirch/defaults/DefaultExternalGetter.scala (hunk 1 above)
Comment (Carlos-Augusto) [reply sdk/src/main/scala/com/ubirch/defaults/DefaultExternalGetter.scala:22]: Agreed. Moved it lower into the pipeline.

File: src/main/scala/com/ubirch/services/state/ExternalStateGetter.scala (hunk 3 above)
Comment (Carlos-Augusto) [reply src/main/scala/com/ubirch/services/state/ExternalStateGetter.scala:26]: Moved it lower into the pipeline.

**Pull request merged**